emulator/src/kreator/assembler.rs

```diff
@@ -0,0 +1,101 @@
+use std::fmt;
+
+struct Assembler {
+    code: Vec<String>,
+    program_counter: u16,
```

Comment:
Throughout the project, pc was used as the identifier for program_counters, so we should stick to that. Also, the pc can be a usize in the context of the assembler as its only used as an index that increases in steps by one.